### PR TITLE
Move solution to root of repository

### PR DESCRIPTION
**Pull request by Clockwork-Muse**
The solution file is in a subdirectory, meaning it builds projects in sibling directories.
A more appropriate place would be the root of the repository.

### CODE REVIEW

**Review by c72578 — COMMENTED** [review]
Please preserve the line endings (Windows CR LF) of the following files:
`collect_files.bat`
`collect_files_debug.bat`

File: README.md [hunk 1]
```diff
@@ -7,21 +7,21 @@ CUERipper is a utility for extracting digital audio from CDs, an open-source alt
...
 Prebuilt binaries can be downloaded from [CUETools Download](http://cue.tools/wiki/CUETools_Download).
 ## Installing from sources
-* Get the CUETools sources from GitHub ([https://github.com/gchudov/cuetools.net](https://github.com/gchudov/cuetools.net)):  
+* Get the CUETools sources from GitHub ([https://github.com/gchudov/cuetools.net](https://github.com/gchudov/cuetools.net)):
 `git clone https://github.com/gchudov/cuetools.net.git`
-* Get the required submodules using:  
+* Get the required submodules using:
 `git submodule update --init --recursive`
-* Apply patches to ThirdParty modules:  
-`git apply --directory=ThirdParty/flac ThirdParty/submodule_flac_CUETools.patch --whitespace=nowarn`
```
Comment (c72578) [inline README.md:15]: Please do not remove the two spaces at the end of the lines. They are responsible for the line break in Markdown [1].
You can see, how README.md looks, e.g. by clicking on `... - View file`.

It currently looks like this in the PR:
Markdown_missing_line_breaks

[1] https://www.markdownguide.org/basic-syntax/#line-breaks

File: GPL.txt [hunk 2]
```diff
@@ -1,340 +1,340 @@
...
-
-  <signature of Ty Coon>, 1 April 1989
-  Ty Coon, President of Vice
-
-This General Public License does not permit incorporating your program into
-proprietary programs.  If your program is a subroutine library, you may
-consider it more useful to permit linking proprietary applications with the
-library.  If this is what you want to do, use the GNU Library General
-Public License instead of this License.
+		    GNU GENERAL PUBLIC LICENSE
```
Comment (c72578) [inline GPL.txt:1]: Please avoid unnecessary white-space changes (line endings).

File: License.txt [hunk 3]
```diff
@@ -1,229 +1,229 @@
...
-IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
-ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
-FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
-DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
-SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
-CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
-OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-********************************************************************************
+********************************************************************************
```
Comment (c72578) [inline License.txt:1]: Please avoid unnecessary white-space changes (line endings).

File: README.md (hunk 1 above)
Comment (Clockwork-Muse) [reply README.md:15]: Sorry about this.  I have "trim whitespace" on by default.

Comment (Clockwork-Muse) [reply README.md:15]: To make this "preservation proof", should we use `<br>` instead?

Comment (c72578) [reply README.md:15]: Please leave the trailing spaces for the line breaks. They are wide-spread in Markdown files.

**Comment by Clockwork-Muse** [comment]
@c72578 - Done.

Sorry it took a while, came down with something.

**Comment by c72578** [comment]
Please also update
`ThirdParty\MAC_SDK\Source\Projects\VS2022\MACLibDll\MACLibDll.vcxproj`
It is still one directory level below the solution dir, e.g.:
`$(SolutionDir)..\obj\`

https://github.com/gchudov/cuetools.net/blob/4d06471a58a2813b97be4dbba8e57e97114b5da7/ThirdParty/MAC_SDK/Source/Projects/VS2022/MACLibDll/MACLibDll.vcxproj#L72

**Comment by Clockwork-Muse** [comment]
@c72578 - Done

File: ThirdParty/MAC_SDK/Source/Projects/VS2022/MACLibDll/MACLibDll.vcxproj [hunk 4]
```diff
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (c72578) [inline ThirdParty/MAC_SDK/Source/Projects/VS2022/MACLibDll/MACLibDll.vcxproj:1]: Please restore the BOM.

**Comment by Clockwork-Muse** [comment]
Should be fixed again.

At some point, though, it would probably be a good idea to remove the BOM/use just `lf` line endings, especially for any move to Linux.  Separate PR just focused on that, though, not something appropriate for this one.

**Comment by c72578** [comment]
TO-DO:

Update location of "CUETools/packages/Newtonsoft.Json.13.0.1/"

```sh
git mv CUETools/packages/ .

git status
        renamed:    CUETools/packages/Newtonsoft.Json.13.0.1/LICENSE.md -> packages/Newtonsoft.Json.13.0.1/LICENSE.md
        renamed:    CUETools/packages/Newtonsoft.Json.13.0.1/Newtonsoft.Json.13.0.1.nupkg -> packages/Newtonsoft.Json.13.0.1/Newtonsoft.Json.13.0.1.nupkg
```

**Comment by Clockwork-Muse** [comment]
@c72578 - Done.

Question, though: why is the package being saved in the source tree, instead of just pulled in as a normal dependency?

**Comment by c72578** [comment]
> Question, though: why is the package being saved in the source tree, instead of just pulled in as a normal dependency?

Possible explanation: `Newtonsoft.Json` is required for building CUETools. It is needed for reading and writing settings. Having the package in the repo ensures independence of package availability.

**Comment by c72578** [comment]
Thanks for the PR.

**Comment by Clockwork-Muse** [comment]
> Possible explanation: `Newtonsoft.Json` is required for building CUETools. It is needed for reading and writing settings. Having the package in the repo ensures independence of package availability.

While that's true, that's not really that much or likely of a problem.

NuGet, like pretty much every package manager I'm aware of, caches packages on the client.  So long as first-build (or at minimum calling `dotnet restore`) is called while online - and recall you'd have to be online to `git clone` in the first place - subsequent builds don't need to be online.

Now, it's true NuGet going down would be a problem in this case... but NuGet going down would be pretty catastrophic in general, and there are ways to set up additional/backup feeds if that were a serious concern.

**Comment by c72578** [comment]
> > > Question, though: why is the package being saved in the source tree, instead of just pulled in as a normal dependency?

> > Possible explanation: `Newtonsoft.Json` is required for building CUETools. It is needed for reading and writing settings. Having the package in the repo ensures independence of package availability.
> 
> While that's true, that's not really that much or likely of a problem.
> 
> NuGet, like pretty much every package manager I'm aware of, caches packages on the client. So long as first-build (or at minimum calling `dotnet restore`) is called while online - and recall you'd have to be online to `git clone` in the first place - subsequent builds don't need to be online.
> 
> Now, it's true NuGet going down would be a problem in this case... but NuGet going down would be pretty catastrophic in general, and there are ways to set up additional/backup feeds if that were a serious concern.

The `Newtonsoft.Json` package will be removed from the source tree part of #277

**Pull request merged**